server/src/internalClusterTest/java/org/elasticsearch/snapshots/RepositoriesIT.java

```diff
@@ -225,4 +229,106 @@ public void testRepositoryVerification() {
             assertThat(ExceptionsHelper.stackTrace(ex), containsString("is not shared"));
         }
     }
+
+    public void testSnapshotDeleteWithRepositoryOutage() throws Exception {
+        disableRepoConsistencyCheck("This test expects residual files in repository");
+        Client client = client();
+        Path repo = randomRepoPath();
+        logger.info("-->  creating repository at {}", repo.toAbsolutePath());
+        assertAcked(client.admin().cluster().preparePutRepository("test-repo")
+            .setType("mock").setSettings(Settings.builder()
+                .put("location", repo)
+                .put("compress", false)
+                .put("chunk_size", randomIntBetween(100, 1000), ByteSizeUnit.BYTES)));
+
+        logger.info("--> creating index-1 and ingest data");
+        createIndex("test-idx-1");
+        ensureGreen();
+        for (int j = 0; j < 10; j++) {
+            indexDoc("test-idx-1", Integer.toString( 10 + j), "foo", "bar" +  10 + j);
+        }
+        refresh();
+
+        // Make repository to throw exception when trying to delete stale indices
+        String masterNode = internalCluster().getMasterName();
+        ((MockRepository)internalCluster().getInstance(RepositoriesService.class, masterNode).repository("test-repo"))
+            .setThrowExceptionWhileDelete(true);
+
+        logger.info("--> creating snapshot");
+        CreateSnapshotResponse createSnapshotResponse = client.admin().cluster().prepareCreateSnapshot("test-repo", "test-snap-1")
```

Comment:
We have `org.elasticsearch.snapshots.AbstractSnapshotIntegTestCase#createFullSnapshot` now which does all of the snapshot creating + logging and assertions here in one call which you can use everywhere instead of this pattern :)

Reply:
Thanks @original-brownbear for suggesting. Updates the test code with ```org.elasticsearch.snapshots.AbstractSnapshotIntegTestCase#createFullSnapshot``` .